React/tic-tac-toe/src/redux/login/reducer.js

```diff
@@ -1,39 +1,19 @@
 import immutable from 'seamless-immutable';
+import { createReducer, completeReducer, completeState } from 'redux-recompose';
 
 import { actions } from './actions';
 
-const initialState = {
-  pending: false,
-  logged: false,
-  authError: ''
+const stateDescription = {
+  user: ''
 };
 
-const reducer = (state = immutable(initialState), action) => {
-  switch (action.type) {
-    case actions.LOGIN_REQUEST:
-      return immutable(state).merge({
-        pending: true,
-        authError: ''
-      });
-    case actions.LOGIN_SUCCESS:
-      return immutable(state).merge({
-        logged: true,
-        pending: false
-      });
-    case actions.LOGIN_FAILURE:
-      return immutable(state).merge({
-        logged: false,
-        pending: false,
-        authError: action.payload
-      });
-    case actions.LOGOUT:
-      return immutable(state).merge({
-        logged: false,
-        pending: false
-      });
-    default:
-      return state;
+const initialState = completeState(stateDescription);
+
+const reducerDescription = {
+  primaryActions: [actions.LOGIN],
+  override: {
+    [actions.LOGOUT]: state => immutable(state).merge({ user: '', userLoading: false })
```

Comment:
let's try this last thing to solve this: `() => ({ ...initialState })`

Reply:
yes, it works :)